pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -200,6 +200,15 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		logger.Error(err, "Error add watching application GVR")
 	}
 
+	if sbr.GetDeletionTimestamp() != nil && sbr.GetOwnerReferences() != nil {
+		logger := logger.WithName("Deleting SBR when it has ownerReference")
+		logger.Debug("Removing resource finalizers...")
+		sbr.SetFinalizers(removeStringSlice(sbr.GetFinalizers(), finalizer))
+		if _, err := updateServiceBindingRequest(r.dynClient, sbr); err != nil {
+			return noRequeue(err)
```

Comment:
In the case the update doesn't go through for any reason, the SBR should be requeued.

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/551/commits/aea0e8b950ab486e265f6d14deee8b8dd4859d9f